api/v2/views/access_token.py

```diff
@@ -0,0 +1,37 @@
+from rest_framework import status
+from rest_framework.response import Response
+
+from core.models import AccessToken, AtmosphereUser
+from core.models.access_token import create_access_token
+
+from api.exceptions import invalid_auth
+from api.v2.serializers.details import AccessTokenSerializer
+from api.v2.views.base import AuthModelViewSet
+
+class AccessTokenViewSet(AuthModelViewSet):
+
+    """
+    API endpoint that allows AccessTokens to be viewed or edited.
+    """
+    serializer_class = AccessTokenSerializer
+
+    def get_queryset(self):
+        """
+        Filter projects by current user.
+        """
+        return AccessToken.objects.filter(token__user=self.request.user)
+
+    def create(self, request):
+        AccessTokenSerializer().validate(request.data)
```

Comment:
Try replacing this line with:
```
++        serializer = self.get_serializer(data=request.data)
++        serializer.is_valid(raise_exception=True)
```

Your class is implementing an AuthModelViewSet, which is based off of rest_framework's ModelViewSet, which provides a default implementation for `def create`. If you're going to override it, you just need to know what all its supposed to do. It basically takes `serializer_class` above does some shit and returns a json repsonse from the serializers data. You don't have to implement the serializers validate method. That method is based on the serializers fields (which in turn are based on the fields defined on your model).

In a manage.py shell run:
```
from api.v2.serializers.details import AccessTokenSerializer
AccessTokenSerializer()
```

Reply:
A helpful trick when you're replacing a method:
```
def create(self, request):
    import ipdb; ipdb.set_trace()
    super(self.__class__, self).create(request)
```
That will call the superclass implementation and you can step through and see what it does.